src/com/palmergames/bukkit/towny/listeners/TownyPlayerListener.java

```diff
@@ -656,6 +656,27 @@ public void onPlayerTeleport(PlayerTeleportEvent event) {
 			}
 		}
 		
+		// Cancel teleport if resident is outlawed in Town
+		if (resident != null && !TownySettings.canOutlawsTeleportOutOfTowns()) {
+			TownBlock tb = TownyAPI.getInstance().getTownBlock(event.getFrom());
+			if (tb != null && tb.hasTown()) {
+				Town town = tb.getTownOrNull();
+				
+				if (town != null && town.hasOutlaw(resident)) {
+					if ((event.getCause() == TeleportCause.COMMAND)) {
+						TownyMessaging.sendErrorMsg(event.getPlayer(), Translation.of("msg_err_outlawed_players_no_teleport"));
+						event.setCancelled(true);
+						return;
+					}
+					if (event.getCause() == TeleportCause.PLUGIN)
+						return;
+					if (!TownySettings.canOutlawsUseTeleportItems() && (event.getCause() == TeleportCause.ENDER_PEARL || event.getCause() == TeleportCause.CHORUS_FRUIT)) {
+						TownyMessaging.sendErrorMsg(event.getPlayer(), Translation.of("msg_err_outlawed_players_no_teleport"));
+						event.setCancelled(true);
+					}
```

Comment:
I think on most servers chorus fruit and ender pearls are in the item_use list and most of the time outsiders are going to not be able to use either item anyways, but for the servers which do not have them there, this would cover it.

Reply:
Gotcha, yeah I thought someone would want this so I added it in